packages/react-router-dom/docs/api/Focus.md

```diff
@@ -0,0 +1,69 @@
+# &lt;Focus>
+
+Provides a way for an application to add [focus management](https://developers.google.com/web/fundamentals/accessibility/focus/using-tabindex#managing_focus_at_the_page_level) after navigation for better accessibility.
+
+```jsx
+import { Focus } from 'react-router-dom'
+
+<Focus>
+  {ref => (
+    <main tabIndex={-1} ref={ref}>
+      {/* ... */}
+    </main>
+  )}
+</Focus>
+```
+
+`Focus` uses a render prop to provide a `ref`. The `ref` should be passed to the element that will be focused.
+
+In order for `Focus` to work, the component type for the focused element needs to either be natively focusable (like an `<input>` or a `<button>`) or be given a `tabIndex` of `-1`. If you do not do this, then the document's `<body>` will be focused instead.
```

Comment:
If an element isn't natively focusable, could we perhaps do a `cloneElement` and add the `tabIndex="-1"` automatically behind the scenes?

Reply:
I don't think that `cloneElement` would work with this API because we don't necessarily have access to the React element. We do have a `ref` to a DOM element, so the attribute could be to the element with `ref.setAttribute("tabindex", "-1")`.

My personal preference would be to require the user to set the tab index. It's a little more work for the dev than RR injecting the attribute, but the `ref` warning will let them know if there is an issue with their code and it lacks any "magic" (being able to focus an element that shouldn't be focusable).

End of the day, (assuming setting the attribute works like I think it would) I can make the switch if you would prefer.